native/src/components/EastereggImage.tsx

```diff
@@ -29,75 +24,57 @@ const ApiUrlText = styled.Text`
   color: red;
 `
 
-class EastereggImage extends React.Component<PropsType, StateType> {
-  constructor(props: PropsType) {
-    super(props)
-    this.state = {
-      clickCount: 0,
-      apiUrlOverride: null,
-      clickStart: null
-    }
-  }
+const EastereggImage = ({ clearResourcesAndCache, theme }: PropsType): ReactElement => {
+  const [clickCount, setClickCount] = useState(0)
+  const [apiUrlOverride, setApiUrlOverride] = useState<string | null>(null)
+  const [clickStart, setClickStart] = useState<null | Moment>(null)
 
-  componentDidMount(): void {
+  useEffect(() => {
     const appSettings = new AppSettings()
     appSettings
       .loadApiUrlOverride()
-      .then(apiUrlOverride =>
-        this.setState({
-          apiUrlOverride
-        })
-      )
+      .then(setApiUrlOverride)
       .catch(e => {
         console.error(e)
       })
-  }
+  }, [])
 
-  onImagePress = async (): Promise<void> => {
+  const onImagePress = useCallback(async () => {
```

Comment:
i'm just wondering why we need here a useCallback hook.
Just wanted to know your idea behind it in this use case